app/services/UserService.scala

```diff
@@ -31,12 +31,14 @@ class UserService @Inject() (implicit val connection: DBConnection) {
     Users.searchUsers(filters)
   }
 
-  def getOrCreateUserByEmail(name: String, familyName: String, email: String): Future[Int] = {
+  def getOrCreateUserByEmail(name: String, familyName: String, email: String): Future[Option[User]] = {
     Users.getUserIdByEmail(email).flatMap {
       case Some(id: Int) =>
         Future.successful(id)
       case _ =>
         Users.add(User(None, name, familyName, email, AccessLevel.Basic))
+    }.flatMap{ id =>
+      getUserById(id)
```

Comment:
It is two extra calls to the db to get that data that we already have in the controller. From the performance point of view I think we should not do this.  

In the controller we can have:
for {
.......
id <- userService.getOrCreateUserByEmail(gUser.givenName, gUser.familyName, gUser.email)
user <- (User(Some(id), gUser.givenName, gUser.familyName, gUser.email, AccessLevel.Basic)
} yield Ok(Json.obj("user" -> Json.toJson(user)))

This way we avoid the two extra call. I know we have a small user base and probably it won't matter, but still, I think if a call to db can be avoided - let's avoid it.

IMO we can reduce the getUserById to one call, but still :)